frontend/pages/hosts/HostDetailsPage/SoftwareTab/SoftwareTab.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { useEffect, useState } from "react";
+import { useDebouncedCallback } from "use-debounce/lib";
+
+import { ISoftware } from "interfaces/software";
+import { VULNERABLE_DROPDOWN_OPTIONS } from "utilities/constants";
+
+// @ts-ignore
+import Dropdown from "components/forms/fields/Dropdown";
+import TableContainer from "components/TableContainer";
+
+import EmptySoftware from "./EmptySoftware";
+import SoftwareVulnCount from "./SoftwareVulnCount";
+
+import generateSoftwareTableHeaders from "./SoftwareTableConfig";
+
+const baseClass = "host-details";
+
+interface ISoftwareTableProps {
+  isLoading: boolean;
+  software: ISoftware[];
+}
+
+const SoftwareTable = ({
+  isLoading,
+  software,
+}: ISoftwareTableProps): JSX.Element => {
+  const [filterName, setFilterName] = useState("");
+  const [filterVuln, setFilterVuln] = useState(false);
+  const [filters, setFilters] = useState({
+    name: filterName,
+    vulnerabilities: filterVuln,
+  });
+
+  useEffect(() => {
+    setFilters({ name: filterName, vulnerabilities: filterVuln });
+  }, [filterName, filterVuln]);
+
+  const onQueryChange = useDebouncedCallback(
+    ({ searchQuery }: { searchQuery: string }) => {
+      setFilterName(searchQuery);
+    },
+    300
+  );
+
+  const onVulnFilterChange = (value: boolean) => {
+    setFilterVuln(value);
+  };
+
+  const renderVulnFilterDropdown = () => {
+    return (
+      <Dropdown
+        value={filters.vulnerabilities}
+        className={`${baseClass}__vuln_dropdown`}
+        options={VULNERABLE_DROPDOWN_OPTIONS}
+        searchable={false}
+        onChange={onVulnFilterChange}
+      />
+    );
+  };
+
+  const tableHeaders = generateSoftwareTableHeaders();
+
+  return (
+    <div className="section section--software">
+      <p className="section__header">Software</p>
+
+      {software.length === 0 ? (
```

Comment:
Is the `=== 0` necessary or would `software.length` by itself work?

Reply:
Also, if you checked on the true operator here `software.length ? ` you could return `TableContainer` first as the true condition, which is the primary element that will be maintained.